client/state/analytics/actions/record.js

```diff
@@ -42,4 +44,6 @@ export const recordPageView = ( url, title, service, properties = {}, options =
 	},
 } );
 
+export const recordPageView = withEnhancers( recordPageViewBase, [ enhanceWithSiteMainProduct ] );
```

Comment:
@tyxla While working on the unit tests, I noticed that there's quite a bit of changes required in order to get the unit tests working. It appears that, the enhanced version of `recordPageView()` can no longer be called directly (at least in the case of the unit tests). 

In the unit `middleware.js` unit tests, what used to be: 
```
dispatch( recordPageView( 'path', 'title', 'default', { name: 'value' } ) )
```
had to be updated to:
```
let pageViewEvent;
const actionDispatcher = ( createdAction ) => ( pageViewEvent = createdAction );
const getState = () => ( { sites: {}, ui: {} } );
const pageViewRecordAction = recordPageView( 'path', 'title', 'default', { name: 'value' } );
pageViewRecordAction( actionDispatcher, getState );

dispatch( pageViewEvent );
```
That said, in the frontend, tracks are being fired as expected. I'm not really familiar with the set up of tracks, so I wasn't sure if the way I'm implementing the additional property raises any concerns.

What I also noticed is that, in cases where the function `recordPageView()` from `client/lib/analytics/page-view.js` is used, the `site_main_product` product will not get added, because tracks aren't triggered by Redux actions in those cases. This does make me wonder, if the original implementation of adding the property to `super-props.js` could be the more straight forward approach that guarantees the addition of the new property. I understand the concern around adding the property to all tracks events, but adding a new property doesn't break the tracking calls, whereas with the current approach, I'm not 100% sure if the change could break existing usage of `recordPageView`. 

What do you think? I'm open to recommendations.

Reply:
I'm really not particularly happy with the idea of polluting all events with an extra prop when it's necessary for only one or even a few events. I'd still prefer having this additional prop separately declared. At the same time, we might want to keep such an action creator as simple and as composable as possible.

So, instead of what we did here and added the enhancer by default, can we just proceed and add this additional enhancer only to the `PageViewTracker` component, just like we used to do with `enhanceWithSiteType`? And if need be, add the `withEnhancers()` call manually where necessary?

We could create a few user-oriented tests then (rendering `PageViewTracker` and testing the results), instead of having to over-complicate the `recordPageView()` tests.

What do you think?

Reply:
Just throwing an idea out there: how about we don't enhance the action at all if the main product is `site`? 

Since the vast majority of WordPress.com sites are *not* domain-only, that should also mean that the vast majority of Tracks calls aren't affected by this PR. Could also mean that we can stick with the current implementation.